src/Model/Behavior/CryptBehavior.php

```diff
@@ -94,29 +94,32 @@ public function beforeSave(Event $event, EntityInterface $entity)
      */
     public function findDecrypted(Query $query, array $options)
     {
-        $options += ['fields' => []];
-        $mapper = function ($row) use ($options) {
-            $driver = $this->_table->connection()->driver();
-            foreach ($this->config('fields') as $field => $type) {
-                if (($options['fields'] && !in_array($field, (array)$options['fields']))
-                    || !$row->has($field)
-                ) {
-                    continue;
+        if(!$query->isEmpty() && $query->isHydrationEnabled()) {
```

Comment:
The query isn't supposed to be executed at this point, so this empty check is too early.